docs/DeveloperGuide.md

```diff
@@ -377,9 +402,262 @@ Use case ends.
+6. TaskManager prompts user to enter changes needed
+7. User writes the change to be made
+8. TaskManager applies the change <br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TaskManager shows an error message <br>
+  Use case resumes at step 2
+
+5a. The given index is invalid
+
+> 5a1. TaskManager shows an error message <br>
+  Use case resumes at step 4
+
+7a. The edit is invalid
+
+> 7a1. TaskManager shows an error message <br>
+  Use case resumes at step 6
+
+8a. User requests to undo the action
+
+> 8a1. TaskManager shows that undo is successful <br>
+  Use case ends
+
+#### Use case: UC04 - Edit tags of a task
+
+**MSS**
+
+1. User requests to list tasks
+2. TaskManager shows a list of tasks
+3. User requests to edit a specific task in the list
+4. TaskManager asks user which section of the task needs changes (description, date/time, or tags)
+5. User selects the tags to be changed
+6. TaskManager shows user list of current tags with extra option to add new one
+7. User selects the index and writes the change to be made
+8. TaskManager applies the change <br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TaskManager shows an error message <br>
+  Use case resumes at step 2
+
+5a. The given index is invalid
+
+> 5a1. TaskManager shows an error message <br>
+  Use case resumes at step 4
+
+7a. The edit entry is invalid
+
+> 7a1. TaskManager shows an error message <br>
+  Use case resumes at step 6
+
+8a. User requests to undo the action
+
+> 8a1. TaskManager shows that undo is successful <br>
+  Use case ends
+
+#### Use case: UC05 - Edit task timing(s)
+
+**MSS**
+
+1. User requests to list tasks
+2. TaskManager shows a list of tasks
+3. User requests to edit a specific task in the list
+4. TaskManager asks user which section of the task needs changes (description, date/time, or tags)
+5. User selects the date/time to be changed
+6. TaskManager prompts user to enter changes needed
+7. User writes the change to be made
+8. TaskManager applies the change <br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TaskManager shows an error message <br>
+  Use case resumes at step 2
+
+5a. The given index is invalid
+
+> 5a1. TaskManager shows an error message <br>
+  Use case resumes at step 4
+
+7a. The edit is invalid
+
+> 7a1. TaskManager shows an error message <br>
+  Use case resumes at step 6
+
+8a. User requests to undo the action
+
+> 8a1. TaskManager shows that undo is successful <br>
+  Use case ends
+
+#### Use case: UC06 - Find a task with a keyword
+
+**MSS**
+
+1. User requests to find task by keyword(s)
+2. TaskManager shows a list of tasks containing given keyword
+3. User selects a task to view in detail.
+4. User can request to make an edit of the task (UC03, UC04, UC05, or UC09)
```

Comment:
this step doesn't belong to this use case. hence can be removed.
instead, you need to mention that after user selects the task, the system will show details of the selected task or something like that.